pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -292,19 +297,36 @@ func (r *Reconciler) resolveRef(ctx context.Context, dm *v1alpha1.DomainMapping)
 
 func (r *Reconciler) reconcileDomainClaim(ctx context.Context, dm *v1alpha1.DomainMapping) error {
 	dc, err := r.netclient.NetworkingV1alpha1().ClusterDomainClaims().Get(ctx, dm.Name, metav1.GetOptions{})
+	if err != nil && !apierrs.IsNotFound(err) {
+		return fmt.Errorf("failed to get ClusterDomainClaim: %w", err)
+	}
+
 	if apierrs.IsNotFound(err) {
-		if dc, err = r.netclient.NetworkingV1alpha1().ClusterDomainClaims().Create(ctx, resources.MakeDomainClaim(dm), metav1.CreateOptions{}); err != nil {
-			return fmt.Errorf("failed to create ClusterDomainClaim: %w", err)
+		dc, err = r.createDomainClaim(ctx, dm)
+		if err != nil {
+			return err
 		}
-	} else if err != nil {
-		return fmt.Errorf("failed to get ClusterDomainClaim: %w", err)
 	}
 
 	if dm.Namespace != dc.Spec.Namespace {
```

Comment:
I'd move this as the `else` clause of `IsNotFound`, since it's meaningless to check it when we just built the `dc` above — we know NS would mathch.